spec/Section 6 -- Execution.md

```diff
@@ -143,6 +147,88 @@ ExecuteMutation(mutation, schema, variableValues, initialValue):
     selection set.
   * Return an unordered map containing {data} and {errors}.
 
+### Subscription
+
+Unlike queries and mutations, subscriptions have a lifetime that consists of three
+phases. Between the the subscribe and unsubscribe phases, the subscription is
+considered to be in the "active" state.
+
+#### Subscribe
+
+The result of executing a subscription operation is a subscription object with
+the following capabilities:
+
+  * **Must** support observation of the associated publish stream (for example,
+  via iteration, callbacks, or reactive semantics).
+  * **Must** support cancellation of the subscription, see {Unsubscribe}.
+  * **Must** support a way to identify the subscriber/subscription pair (for
+    example, a GUID/callback table or closure over the callback).
+  * **May** include an initial response associated with executing the selection
+  set defined on the subscription operation.
+
+Subscribe(schema, subscription, operationName, variableValues, initialValue):
+
+  * Let {subscriptionType} be the root Subscription type in {schema}.
+  * Assert: {subscriptionType} is an Object type.
+  * Let {selectionSet} be the top level Selection Set in {subscription}.
+  * Let {rootField} be the first top level field in {selectionSet}.
+  * Let {eventStream} be the result of running {MapSubscriptionEvents(rootField, variableValues)}.
+  * Optionally: let {data} be the result of running {ExecuteRequest(schema, document, operationName, variableValues, initialValue)}, and return {data} on the subscription object.
```

Comment:
I find this very confusing and probably not what we want. If I understand this correctly, sometimes this algorithm results in a value `{data}` which is returned, sometimes it does not - in which case it returns nothing?

Also, this algorithm does not describe what should be done with `{eventStream}` once created - I would have expected `{eventStream}` to be returned.

As a concrete example - what should I expect to see differently between a subscription which has an initial response vs one that does not, say in terms of websocket payloads? Do I actually get a `{}` or `{data: null}` payload as an "ACK" for subscriptions without an initial value? Or is "ACK" a detail that belongs to the transport mechanism?

Perhaps another way to frame this is that "initial value" could be a first published event which simply occurs instantaneously after the creation of a subscription?